src/components/heartbeats/heartbeatsStore.ts

```diff
@@ -0,0 +1,148 @@
+import { bufferToHex } from 'ethereumjs-util';
+import { observable, reaction, autorun, toJS } from 'mobx';
+import autobind from 'autobind-decorator';
+import { web3PlasmaStore } from '../../stores/web3/plasma';
+import { operatorStore } from '../../stores/operator';
+import storage from '../../utils/storage';
+import { range } from '../../utils/range';
+
+import { Tx, LeapTransaction } from 'leap-core';
+
+type SlotHeartbeatsData = {
+  periods: Set<number>;
+  lastPeriodChecked: number;
+  owner: string;
+};
+
+export class HeartbeatsStore {
+  private storageKey: string;
+
+  @observable
+  private heartbeatColor: number;
+
+  public keepPeriodsLimit: number;
+
+  @observable
+  public latestBlock: number = 0;
+
+  @observable
+  public heartbeats: SlotHeartbeatsData[];
+
+  constructor(keepPeriodsLimit = 15) {
+    this.keepPeriodsLimit = keepPeriodsLimit;
+
+    reaction(
+      () => operatorStore.slots,
+      () => {
+        this.storageKey = `hb-${operatorStore.address}`;
+        this.heartbeats = this.initHeartbeatsData();
+        operatorStore.contract.methods
+          .heartbeatColor()
```

Comment:
Currently this method is not supported on testnet (needs an update of poaContract, I guess).